dvc/scm/git/backend/pygit2.py

```diff
@@ -84,7 +84,7 @@ def is_sha(rev: str) -> bool:
     def dir(self) -> str:
         raise NotImplementedError
 
-    def add(self, paths: Iterable[str]):
+    def add(self, paths: Iterable[str], update=False):
```

Comment:
`add --update` is available in `libgit2`, but `pygit2` does not import the update related index functions. I'll submit the pygit PR for this at some point in the near future.